homeassistant/components/sws12500/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for Sencor SWS 12500 Weather Station integration."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import API_ID, API_KEY, DOMAIN
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(API_ID, default="API ID"): str,
+        vol.Required(API_KEY, default="API KEY"): str,
+    }
+)
+
+
+class CannotConnect(HomeAssistantError):
+    """We can not connect. - not used in push mechanism."""
+
+
+class InvalidAuth(HomeAssistantError):
+    """Invalid auth exception."""
+
+
+class ConfigOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle WeatherStation options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+
+        api_id = self.config_entry.options.get(API_ID)
+        api_id = api_id if api_id else self.config_entry.data.get(API_ID)
+
+        api_key = self.config_entry.options.get(API_KEY)
+        api_key = api_key if api_key else self.config_entry.data.get(API_KEY)
+
+        OPTIONAL_USER_DATA_SCHEMA = vol.Schema(  # pylint: disable=invalid-name
+            {
+                vol.Required(API_ID, default=api_id): str,
+                vol.Required(API_KEY, default=api_key): str,
+            }
+        )
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="init", data_schema=OPTIONAL_USER_DATA_SCHEMA, errors=errors
+            )
+
+        return self.async_create_entry(title=DOMAIN, data=user_input)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Sencor SWS 12500 Weather Station."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            await self.async_set_unique_id(DOMAIN)
+            self._abort_if_unique_id_configured()
+
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        if user_input[API_ID] != "" and user_input[API_KEY] != "":
+            return self.async_create_entry(title=DOMAIN, data=user_input)
```

Comment:
You are checking whether user_input is empty lines, but your default values for the data scheme are `API ID` and `API KEY` strings. So if users do not change the defaults, they will set up the integration which won't work, isn't it?